Add support for device id argument to support multiple 4 and 8 relay devices

dirck opened the pull request:
Added support for multiple 4 & 8 devices using the device id argument.  This can be the literal serial number or can include '#'+index number , so:

    testCommon.TestCommon(dae_RelayBoard.DAE_RELAYBOARD_TYPE_8, 'DAE#0')

and

    testCommon.TestCommon(dae_RelayBoard.DAE_RELAYBOARD_TYPE_8, 'DAE#1')

Notes:
*  FT2DXXLinux.py only, didn't even look at the Windows module
* The diff includes lots of white space only changes: my editor trims trailing spaces
* just stripped a print statement, you might want it back 
* sources in dae-py-relay-controller have mixed ends of lines.  Fixing this requires adding .gitattributes for auto text, which will result in lots of non-functional changes.  Here's the info:

dae_RelayBoard/dae_RelayBoard_Common.py: 0 <CR> and 30 <LF>: unix.
dae_RelayBoard/dae_RelayBoard_D2XX.py: 139 <CR> and 139 <LF>: windows.
dae_RelayBoard/dae_RelayBoard.py: 98 <CR> and 98 <LF>: windows.
dae_RelayBoard/dae_RelayBoard_VCP.py: 169 <CR> and 169 <LF>: windows.
dae_RelayBoard/FTD2XXDllWrap.py: 0 <CR> and 158 <LF>: unix.
dae_RelayBoard/FTD2XXLinux.py: 0 <CR> and 61 <LF>: unix.
dae_RelayBoard/FTD2XXWindows.py: 0 <CR> and 60 <LF>: unix.
dae_RelayBoard/__init__.py: 0 <CR> and 0 <LF>= no end-of-lines
Example & Test/example16.py: 0 <CR> and 75 <LF>: unix.
Example & Test/example4.py: 75 <CR> and 75 <LF>: windows.
Example & Test/example8.py: 74 <CR> and 74 <LF>: windows.
Example & Test/__init__.py: 0 <CR> and 0 <LF>= no end-of-lines
Example & Test/test16Intervals.py: 0 <CR> and 42 <LF>: unix.
Example & Test/test16.py: 0 <CR> and 28 <LF>: unix.
Example & Test/test4.py: 0 <CR> and 25 <LF>: unix.
Example & Test/test8.py: 0 <CR> and 25 <LF>: unix.
Example & Test/testCommon.py: 100 <CR> and 100 <LF>: windows.
Example & Test/testExceptions.py: 55 <CR> and 55 <LF>: windows.

petersbingham commented on dae_RelayBoard/FTD2XXLinux.py:
```diff
@@ -23,39 +23,53 @@
+
     def initialise(self, deviceID, baudRate, mask, bitMode):
         try:
+            found = 0
```
Can you please move found and index to the block of code just before they are used.

petersbingham commented on dae_RelayBoard/FTD2XXLinux.py:
```diff
@@ -23,39 +23,53 @@
-            self.bb.open()
+                    if which == index:
+                        found = 1
+                        d2 = dev[2].decode()
```
There is no need for d2 here, as far as I can see.

petersbingham commented on dae_RelayBoard/FTD2XXLinux.py:
```diff
@@ -23,39 +23,53 @@
-            self.bb.direction = mask
-            self.bb.open()
+                    if which == index:
+                        found = 1
```
Can you please use boolean True False (at at point of initialisation above).

petersbingham commented on dae_RelayBoard/FTD2XXLinux.py:
```diff
@@ -23,39 +23,53 @@
+                        found = 1
+                        d2 = dev[2].decode()
+                        # DEBUG
+                        #print('Found device %s#%d => %s' % (deviceID, index, d2))
```
Yes, please leave this in place.

petersbingham commented:
Thanks for this. Some minor inline comments left. Yes, good to remove whitespace. If you want you can add in the .gitattributes as a separate PR, with all resultant changes.

Can you please squash all of your commits.
Once above has been addressed I will test and accept.

petersbingham commented:
I think you should be able to squash and force push. I haven't made any changes lately so no rebase should be required.

dirck commented:
I used rebase to squash locally and then used push --force.  A couple of minor additional changes.  Added:

    from __future__ import print_function

to dae_RelayBoard/FTD2XXLinux.py  for Python 2 / 3 compatibility (vs actually printing a tuple in Python 2), and added a couple of indexed examples (comments only!) in test8.py.
